### PR TITLE
overlay groupBy method to support array

### PR DESCRIPTION
**Pull request by nibsirahsieu**

### CODE REVIEW

**Comment by nibsirahsieu** [comment]
mmhh.. test is failed for all db=agnostic.. anyone know how to fix it?

File: tests/Propel/Tests/Runtime/ActiveQuery/ModelCriteriaGroupByArrayTest.php [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ */
+
+namespace Propel\Tests\Runtime\ActiveQuery;
+
+use Propel\Tests\Helpers\Bookstore\BookstoreEmptyTestBase;
+use Propel\Tests\Bookstore\Book;
+use Propel\Tests\Bookstore\Author;
+use Propel\Tests\Bookstore\AuthorQuery;
+
+class ModelCriteriaGroupByArrayTest extends BookstoreEmptyTestBase
```
Comment (marcj) [inline tests/Propel/Tests/Runtime/ActiveQuery/ModelCriteriaGroupByArrayTest.php]: you're having in this class database access, so you need to mark this class as "@database".

```php
/**
 * @group database
 */
class ModelCriteriaGroupByArrayTest extends BookstoreEmptyTestBase
```

**Comment by nibsirahsieu** [comment]
thank @marcj for the hint.

**Comment by marcj** [comment]
Ok, cool! could you also please squash/rebase your commits into one? Then I'd merge this :)

**Comment by nibsirahsieu** [comment]
done

**Comment by nibsirahsieu** [comment]
anything else before it's get merged? I will submit another PR to overlay `orderBy` function if you agree. Thanks

**Comment by marcj** [comment]
Great, sorry for the delay. very nice contribution, thanks :)

**Pull request merged**